Update tableaudesktop.sh

PacketPalantir opened the pull request:
All questions must be filled out or your Pull Request will be closed for lack of information. The first three questions should be answered `Yes` before submitting the pull request.
---
**Have you confirmed this pull request is not a duplicate?**
Yes

**Is this pull request creating or modifying a label in the fragments/labels folder, and not Installomator.sh itself?**
Yes

**Did you use [our editorconfig file](https://github.com/Installomator/Installomator/wiki/Contributing-to-Installomator)?**
Yes

**Additional context** Add any other context about the label or fix here.

Tableau Desktop has a separate download for Apple Silicon (arm64). Added `https://www.tableau.com/downloads/desktop/mac-arm64` to `tableaudesktop.sh`

**Installomator log** At the bottom of this pull request, provide a log of a label run by running Installomator in Terminal and saving the output. `DEBUG=1` can be enabled but **do not enable [Debug logging level](https://github.com/Installomator/Installomator/wiki/Configuration-and-Variables#logging-level) and please format the log [using a code block](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks#fenced-code-blocks)!**

```
2025-08-09 13:38:18 : INFO  : tableaudesktop : setting variable from argument DEBUG=0
2025-08-09 13:38:18 : INFO  : tableaudesktop : Total items in argumentsArray: 1
2025-08-09 13:38:18 : INFO  : tableaudesktop : argumentsArray: DEBUG=0
2025-08-09 13:38:18 : REQ   : tableaudesktop : ################## Start Installomator v. 10.9beta, date 2025-08-09
2025-08-09 13:38:18 : INFO  : tableaudesktop : ################## Version: 10.9beta
2025-08-09 13:38:18 : INFO  : tableaudesktop : ################## Date: 2025-08-09
2025-08-09 13:38:18 : INFO  : tableaudesktop : ################## tableaudesktop
2025-08-09 13:38:18 : INFO  : tableaudesktop : SwiftDialog is not installed, clear cmd file var
2025-08-09 13:38:19 : INFO  : tableaudesktop : Reading arguments again: DEBUG=0
2025-08-09 13:38:19 : INFO  : tableaudesktop : BLOCKING_PROCESS_ACTION=tell_user
2025-08-09 13:38:19 : INFO  : tableaudesktop : NOTIFY=success
2025-08-09 13:38:19 : INFO  : tableaudesktop : LOGGING=INFO
2025-08-09 13:38:19 : INFO  : tableaudesktop : LOGO=/System/Applications/App Store.app/Contents/Resources/AppIcon.icns
2025-08-09 13:38:19 : INFO  : tableaudesktop : Label type: pkgInDmg
2025-08-09 13:38:19 : INFO  : tableaudesktop : archiveName: Tableau Desktop.dmg
2025-08-09 13:38:19 : INFO  : tableaudesktop : no blocking processes defined, using Tableau Desktop as default
2025-08-09 13:38:19 : INFO  : tableaudesktop : No version found using packageID com.tableausoftware.tableaudesktop
2025-08-09 13:38:19 : INFO  : tableaudesktop : name: Tableau Desktop, appName: Tableau Desktop.app
2025-08-09 13:38:19 : WARN  : tableaudesktop : No previous app found
2025-08-09 13:38:19 : WARN  : tableaudesktop : could not find Tableau Desktop.app
2025-08-09 13:38:19 : INFO  : tableaudesktop : appversion: 
2025-08-09 13:38:19 : INFO  : tableaudesktop : Latest version of Tableau Desktop is 2025.2.1.
2025-08-09 13:38:19 : REQ   : tableaudesktop : Downloading https://www.tableau.com/downloads/desktop/mac-arm64 to Tableau Desktop.dmg
2025-08-09 13:38:30 : INFO  : tableaudesktop : Downloaded Tableau Desktop.dmg – Type is  bzip2 compressed data, block size = 100k – SHA is db7b9cf89cf196c89f3478a4a562f2cdf043550b – Size is 755604 kB
2025-08-09 13:38:30 : REQ   : tableaudesktop : no more blocking processes, continue with update
2025-08-09 13:38:30 : REQ   : tableaudesktop : Installing Tableau Desktop
2025-08-09 13:38:30 : INFO  : tableaudesktop : Mounting /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.YZPuUPtesM/Tableau Desktop.dmg
2025-08-09 13:39:01 : INFO  : tableaudesktop : Mounted: /Volumes/Tableau Desktop
2025-08-09 13:39:01 : INFO  : tableaudesktop : found pkg: /Volumes/Tableau Desktop/Tableau Desktop.pkg
2025-08-09 13:39:01 : INFO  : tableaudesktop : Verifying: /Volumes/Tableau Desktop/Tableau Desktop.pkg
2025-08-09 13:39:01 : INFO  : tableaudesktop : Team ID: QJ4XPRK37C (expected: QJ4XPRK37C )
2025-08-09 13:39:01 : INFO  : tableaudesktop : Installing /Volumes/Tableau Desktop/Tableau Desktop.pkg to /
2025-08-09 13:39:41 : INFO  : tableaudesktop : Finishing...
2025-08-09 13:39:44 : INFO  : tableaudesktop : No version found using packageID com.tableausoftware.tableaudesktop
2025-08-09 13:39:44 : INFO  : tableaudesktop : name: Tableau Desktop, appName: Tableau Desktop.app
2025-08-09 13:39:44 : INFO  : tableaudesktop : App(s) found: /Applications/Tableau Desktop (Apple silicon) 2025.2.app
2025-08-09 13:39:44 : INFO  : tableaudesktop : found app at /Applications/Tableau Desktop (Apple silicon) 2025.2.app, version 2025.2.1, on versionKey CFBundleShortVersionString
2025-08-09 13:39:44 : REQ   : tableaudesktop : Installed Tableau Desktop, version 2025.2.1.
2025-08-09 13:39:44 : INFO  : tableaudesktop : notifying
displaynotification:7: no such file or directory: /usr/local/bin/dialog
2025-08-09 13:39:45 : INFO  : tableaudesktop : Installomator did not close any apps, so no need to reopen any apps.
2025-08-09 13:39:45 : REQ   : tableaudesktop : All done!
2025-08-09 13:39:45 : REQ   : tableaudesktop : ################## End Installomator, exit code 0 
```

Please identify any issues fixed by your pull request by including the issue number. (Example: "Fixes #XXXX")

PacketPalantir committed:
Add arm64 download for Apple Silicon

a-martin253 reviewed (changes requested):
This is almost spot on to what we need this to be updated to in main but the download URL paths have changed slightly.

```
tableaudesktop)
    name="Tableau Desktop"
    type="pkgInDmg"
    packageID="com.tableausoftware.tableaudesktop"
    if [[ $(/usr/bin/arch) == "arm64" ]]; then
        downloadURL="https://www.tableau.com/downloads/desktop/reg-mac-arm64"
    else
        downloadURL="https://www.tableau.com/downloads/desktop/reg-mac"
    fi
    appNewVersion=${$(curl -fsIL "$downloadURL" | sed -nE 's/.*Desktop-([0-9-]*).*/\1/p')//-/.}
    expectedTeamID="QJ4XPRK37C"
    ;;
```

PacketPalantir committed:
Updated Tableau URLs

jschwager commented:
@PacketPalantir appNewVersion does not function correctly with the ARM download. There's an extra period at the end. Try this:
```
    if [[ $(arch) == "arm64" ]]; then
        downloadURL="https://www.tableau.com/downloads/desktop/reg-mac-arm64"
        appNewVersion=${$(curl -fsIL "$downloadURL" | sed -nE 's/.*Desktop-([0-9-]*).*/\1/p' | sed 's/-/./g; s/\.$//')}
    elif [[ $(arch) == "i386" ]]; then
        downloadURL="https://www.tableau.com/downloads/desktop/mac"
        appNewVersion=${$(curl -fsIL "$downloadURL" | sed -nE 's/.*Desktop-([0-9-]*).*/\1/p')//-/.}
    fi
```

uurazzle reviewed (changes requested):
Hi @PacketPalantir,

Thanks for the PR. We noticed the label sets `packageID` on a pkg-based app install.

### Remove unnecessary packageID for app installs

For pkg installers that install an app into `/Applications`, `packageID` is usually unnecessary. Installomator can use the installed app bundle for version checking instead, which avoids relying on package receipts that may remain after an app is removed. The [Label Variables Reference](https://github.com/Installomator/Installomator/wiki/Label-Variables-Reference) notes that `packageID` is recommended for non-app installs, command-line tools, or apps not located in `/Applications`.

Please remove `packageID` if this installer places the app in `/Applications`, and use standard app bundle variables instead when needed:

```bash
appName="MyApp.app"
versionKey="CFBundleVersion"
```

If the pkg does not install an app bundle that Installomator can discover, please add a short note in the PR explaining why `packageID` is required. Thanks!

PacketPalantir committed:
Update tableaudesktop.sh

Removed `packageID` per https://github.com/Installomator/Installomator/pull/2511#pullrequestreview-4222552265

uurazzle reviewed (approved):
Hi @PacketPalantir,

Thanks for contributing this Installomator label ([#2511: Update tableaudesktop.sh](https://github.com/Installomator/Installomator/pull/2511))—everything looks good. Appreciate you taking the time to add it to the project!

gilburns reviewed (approved):
Thank you for your contribution to the project!
